api/client/ffs.go

```diff
@@ -607,7 +607,7 @@ func fromRPCDealErrors(des []*rpc.DealError) ([]ffs.DealError, error) {
 	res := make([]ffs.DealError, len(des))
 	for i, de := range des {
 		var propCid cid.Cid
-		if de.ProposalCid != "" {
+		if de.ProposalCid != "" && de.ProposalCid != "b" {
```

Comment:
So, if the original  value was `cid.Undef`, the stringified result is `"b"`.  Cover in the client this edge case just in case it gets slipped in the RPC layer.